recipe/meta.yaml

```
@@ -7,9 +7,11 @@ package:
 source:
   url: http://www.mlpack.org/files/mlpack-{{ version }}.tar.gz
   sha256: 61c604026d05af26c244b0e47024698bbf150dfcc9d77b64057941d7d64d6cf6
+  patches:
+    - numpy2.patch
```

Yep this is the easiest path with patches. Plus it handles some corner cases with Windows and various types of patch formats

It might help to link the PR in a comment for easier tracking/cleanup

```suggestion
  patches:
    # xref: https://github.com/mlpack/mlpack/pull/3752
    - numpy2.patch
```